### PR TITLE
Fish shell: fix switch-off and off; add version check for switch

### PR DESCRIPTION
**Pull request by mcfog**
+ switch-off and off not working correctly: see phpbrew/phpbrew#860
+ switch: `phpbrew switch UNEXIST_VERSION` would write "Invalid Argument xxx" into $PHPBREW_HOME/init, thus breaking fish boot procedure.

issues reproduced at (phpbrew 1.23.1 + fish 3.0.2) and (phpbrew 1.22.6 + fish 2.7.1)
fix tested on above environments

maybe alse fixes phpbrew/phpbrew#905 phpbrew/phpbrew#926

### CODE REVIEW

**Comment by c9s** [comment]
@racklin I don’t know fish. can you review this fix? Thank you

File: shell/phpbrew.fish [hunk 1]
```diff
@@ -365,7 +372,11 @@ function __phpbrew_reinit
         mkdir -p -p "$PHPBREW_HOME"
     end
     __phpbrew_update_config $_PHP_VERSION
+    if [ $status -ne 0 ]
+        return $status
+    end
     __phpbrew_set_path
+    return 0
```
Comment (morozov) [inline shell/phpbrew.fish]: This could be omitted for brevity.

File: shell/phpbrew.fish [hunk 2]
```diff
@@ -347,6 +348,12 @@ function __phpbrew_update_config
     else
         set VERSION $argv
     end
+
+    if [ ! -d "$PHPBREW_ROOT/php/$VERSION" ]
+        echo "php version: $VERSION not exists."
```
Comment (morozov) [inline shell/phpbrew.fish]: Looks like we're missing the same in `bashrc`. Could you add this logic there as well?

**Comment by c9s** [comment]
@mcfog please check the review, thank you

**Pull request closed without merging**